### DISCUSSION TITLE
Using shared variables in magic commands

### DISCUSSION THREAD

**Discussion by andrerav** [dotnet/interactive#2290] [Q&A] [answered]
Hi!

I wish to connect to a database using #!connect, but instead of specifying the connection string in a literal manner I would like to use a variable that contains a connection string defined earlier in my notebook. So I imagine something like this:

(C#)
```csharp
var connectionString = "....";
```

(SQL)
```sql
#!share --from csharp connectionString 
#!connect mssql --kernel-name someDb connectionString

SELECT * ....
```

But alas, this does not work. Is there some way to share and use the connectionString variable in the #!connect magic command?

**Comment by huhubun** [comment, marked as answer]
I had the same problem and found the answer here https://github.com/dotnet/interactive/issues/2523#issuecomment-1331066338 😀

**Reply by andrerav** [reply]
Perfect, thank you!
